### PR TITLE
upgrade-notes: add information about kvstoremesh and external workloads

### PR DESCRIPTION
**Pull request by marseel**
Related: #33694

### CODE REVIEW

File: Documentation/operations/upgrade.rst [hunk 1]
```diff
@@ -388,6 +388,9 @@ Annotations:
 * KVStoreMesh is now enabled by default in Clustermesh.
   If you want to disable KVStoreMesh, set Helm value ``clustermesh.apiserver.kvstoremesh.enabled=false``
   explicitly during the upgrade.
+* With the default enablement of KVStoreMesh, if you use :ref:`external workloads <external_workloads>`,
+  ensure that your cluster has a Cluster name and ID specified before upgrading.
+  Alternatively, you can opt out of KVStoreMesh.
```
Comment (giorio94) [inline Documentation/operations/upgrade.rst]:
```suggestion
  Alternatively, you need to explicitly opt out of KVStoreMesh.
```

Nit. It seems a bit clearer to me.

**Review by giorio94 — APPROVED** [review]
I wonder whether we should also add a deprecation notice that starting from v1.17 external workloads are no longer supported in combination with ClusterID=0. That combination has already proved to be problematic in the past as it needs to be handled as a special case, and there's the possibility of unexpectedly breaking it now that it is no longer explicitly tested in CI.

File: Documentation/operations/upgrade.rst (hunk 1 above)
Comment (marseel) [reply Documentation/operations/upgrade.rst]: How about "Alternatively, you can explicitly opt out of KVStoreMesh."
"need" doesn't work with "alternatively", at least for me :sweat_smile:

Comment (giorio94) [reply Documentation/operations/upgrade.rst]: Yep, sounds good to me.

**Comment by marseel** [comment]
/test-backport-1.16

**Review by michi-covalent — APPROVED** [review]

**Review by aditighag — APPROVED** [review]

**Pull request merged**